P202.Training.Test/RoleServiceTest.cs

```diff
@@ -0,0 +1,88 @@
+using System.Text;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace P202.Training.Test
+{
+    public class RoleServiceTest
+    {
+        [Fact]
+        public void RoleService_CreateData()
+        {
+            var newRole = new Role();
+            var mockpersonRepository = new Mock<IRoleRepository>();
+            mockpersonRepository.Setup(x => x.CreateRole(newRole)); 
+            mockpersonRepository.Object.CreateRole(newRole);
+            mockpersonRepository.Verify(x => x.CreateRole(newRole), Times.Once());
+        }
+
+        [Fact]
+        public void RoleService_DeleteData()
+        {
+            var deleteRole = new Role();
+            var mockpersonRepository = new Mock<IRoleRepository>();
+            mockpersonRepository.Setup(x => x.DeleteRole(deleteRole));
+            mockpersonRepository.Object.DeleteRole(deleteRole);
+            mockpersonRepository.Verify(x => x.DeleteRole(deleteRole), Times.Once());
+        }
+
+        [Fact]
+        public void RoleService_UpdateData()
+        {
+            var updateRole = new Role { Name = "new-role-name" };
+            var mockpersonRepository = new Mock<IRoleRepository>();
+            mockpersonRepository.Setup(x => x.UpdateRole(updateRole)).Returns(updateRole);
+            mockpersonRepository.Object.UpdateRole(updateRole);
+            mockpersonRepository.Verify(x => x.UpdateRole(updateRole), Times.Once());
+        }
+
+        [Fact]
+        public void RoleService_ReadData()
+        {
+            var role = new Role { Id = 3 };
+            var mockroleRepository = new Mock<IRoleRepository>();
+            mockroleRepository.Setup(x => x.GetRole(3)).Returns(role); 
+            mockroleRepository.Object.GetRole(3); 
+            mockroleRepository.Verify(x => x.GetRole(role.Id), Times.Once());
+        }
+
+        [Fact]
+        public void RoleService_ListOfData()
+        {
+            // Arrange
+            Mock<IRoleRepository> roleRepository = new Mock<IRoleRepository>();
+            Mock<IMappingService> mappingService = new Mock<IMappingService>();
+            
+            List<Role> role = new List<Role>();
+            roleRepository.Setup(r => r.GetAllRoles()).Returns(role.ToList<Role>());
+
+            var rolList = new List<Role> {
+                new Role { 
+                    Name = "admin"
+                },
+                new Role {
+                    Name = "salesman"
+                }
+            };
+
+            mappingService.Setup(m => m.Map<IEnumerable<Role>, List<Role>>(It.IsAny<IEnumerable<Role>>()))
+                            .Returns(rolList);
+
+            // Act
+            var result = roleRepository.Object.GetAllRoles();
+            //Assert that the Add method was called once
+            roleRepository.Verify(x => x.GetAllRoles(), Times.Once());
+
+            // Assert
+            var model = Assert.IsType<List<Role>>(result);
+            Assert.Equal(2, model.Count);
+        }
```

Comment:
This test has the right idea about different sections for setup/act/assert. Could you replicate that on the rest of the tests?

Reply:
For sure @DanielAltamirano . I 'll replicate that on the rest of the test as you suggest